lib/rexml/parsers/sax2parser.rb

```diff
@@ -260,10 +260,7 @@ def add( pair )
 
       def get_namespace( prefix )
         return nil if @namespace_stack.empty?
-
-        uris = (@namespace_stack.find_all { |ns| not ns[prefix].nil? }) ||
-          (@namespace_stack.find { |ns| not ns[nil].nil? })
-        uris[-1][prefix] unless uris.nil? or 0 == uris.size
+        @namespace_stack[-1][prefix] || @namespace_stack[-1][nil]
```

Comment:
We can avoid multiple `namespace_stack[-1]` calls here:

```suggestion
        current_namespace = @namespace_stack[-1]
        current_namespace[prefix] || current_namespace[nil]
```

Reply:
OK, I see.